packages/backend/src/services/DashboardService/DashboardService.ts

```diff
@@ -375,6 +420,24 @@ export class DashboardService {
                     DashboardService.getCreateEventProperties(updatedDashboard),
             });
             await this.deleteOrphanedChartsInDashboards(user, dashboardUuid);
+
+            // Track charts created in dashboard
+            const chartsInExistingDashboard =
+                DashboardService.findChartsThatBelongToDashboard(
+                    existingDashboard,
+                );
+            const chartsInUpdatedDashboard =
```

Comment:
I'm not sure I understand why we need this

Reply:
When we update a chart, we get all the chart that belong to the dashboard before the update ( eg [a, b] ) and after the update ( eg: [a, b, c] ) 
From that information, we know what charts were added by comparing the 2 arrays. `c` was not there before the update so it was created.